cabal-install/Distribution/Client/Config.hs

```diff
@@ -873,6 +914,15 @@ configFieldDescriptions src =
         configAllowNewer (\v flags -> flags { configAllowNewer = v })
        ]
 
+  ++ toSavedConfig liftInitFlag
+       (initOptions ParseArgs)
+       ["quiet", "no-comments", "minimal", "overwrite", "package-dir",
```

Comment:
I'm wondering if `toSavedConfig` would be better if this was an explicit list of what to include in the global config instead of an exclude list. In this case `non-interactive` is the only thing (at this point) from `InitFlags` that we want to be able to configure in `~/.cabal/config`. Using an exclude list also introduces the possibility of newly added flags sneaking into the global config. Just a thought.

Reply:
Makes sense. We could also do something like `IncludeAll | IncludeWhiteList [FlagName]`.

Reply:
Added to my TODO list :)